src/bigquery/base_collector.py

```diff
@@ -0,0 +1,33 @@
+# Copyright © 2020 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Author: Dharmendra G Patel <dhpatel@redhat.com>
+#
+"""Base collector class to parse and extract dependencies from manifests."""
+from collections import Counter
+
+
+class BaseCollector:
+    """Base class to handle manifests and extract dependencies."""
+
+    def __init__(self, name):
+        """Contructor for base collector."""
+        self.name = name
+        self.counter = Counter()
+
+    def _update_counter(self, packages=list):
```

Comment:
Did you mean to have an empty list as default value? or it is a type hint?

Reply:
Yes, we need default empty list as we update S3 data using all collectors in loop. So empty list will generate empty DICT in output.

Reply:
> Yes, we need default empty list as we update S3 data using all collectors in loop. So empty list will generate empty DICT in output.
This is incorrect then. You must use `packages=list()`, however this is not a recommended practise as it would create a global list and every caller would be sharing it.

P.S. https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

Reply:
@arajkumar Each derived collector ie., mave, pypi and npm are calling super init to initiate `self.counter` with a new object of `Counter()` there is no global copy shared among them. What you share refers to the default argument initialization, which is not the case in above code.

Reply:
The question is about the packages parameters. I see that you are passing in a list when you call this function. Are you trying to create a type hint?